src/pilot/config_dashboard.py

```diff
@@ -350,254 +293,304 @@ def prep_ceph_hosts(dashboard_node, ceph_nodes):
+
+            # Delete any prior ceph conf file entries
+            try:
+                beg = host_entries.index(beg_banner)
+                end = host_entries.index(end_banner)
+                del(host_entries[beg:end+1])
+            except:
+                pass
+
+            # Create ceph_conf with the preluminous entry
+            with open(tmp_hosts, "w") as f:
+                f.writelines(host_entries)
+                f.write("{}{}\n{}".format(
+                    beg_banner,
+                    "mon_health_preluminous_compat=true",
+                    end_banner))
+
+            # Upload the new file to the node
+            node.put(tmp_hosts, tmp_hosts)
+            node.run("sudo cp {} {}.bak"
+                     .format(Node.ceph_conf, Node.ceph_conf))
+            node.run("sudo mv {} {}".format(tmp_hosts, Node.ceph_conf))
+            node.run("sudo restorecon {}".format(Node.ceph_conf))
+            os.unlink(tmp_hosts)
+
+
+def prep_collectd(dashboard_node, ceph_nodes):
+    LOG.info("Preparing the Ceph Storage Cluster collectd services.")
+    collectd_dir = "/etc/collectd.d"
 
-def start_calamari_server(dashboard_node, calamari_node):
-    """Starts the Calamari server
-
-    Runs the command that initialzes and starts the Calamari server (one
-    of the Ceph Monitor nodes)
-
-    Under the old Calamari UI (prior to the Storage Console), the Calamari
-    server ran on the Ceph admin VM, and was configured with the username
-    and password used to log into the UI.
-
-    With the Storage Console, the Calamari server runs on one of the Ceph
-    Monitor nodes, and it communicates with the Storage Console UI running
-    on a VM. In fact, the Storage Console is the one that logs into the
-    Calarmari server.
-
-    NOTE:
-
-    The Storage Console installation instructions mention it uses hard-coded
-    credentials when logging into the Calamari server !!!  It's important the
-    Calamari server be initialed to use the same credentials or else the
-    Storage Console won't be able to connect to it. The hard-coded credentials
-    are... (wait for it!) admin/admin.
+    for node in ceph_nodes:
+        collectd_files = ('network.conf', 'disk.conf')
+        for file in collectd_files:
+            conf_file = os.path.join(os.sep, collectd_dir, file)
+            status, stdout, stderr = node.execute("[ -f {} ] \
+                                                  && echo true \
+                                                  || echo false"
+                                                  .format(conf_file))
+            LOG.debug("STDOUT for node ({}) file ({}) = {}"
+                      .format(node.fqdn, conf_file, stdout))
+            if "true" in stdout:
+                node.run("mv {} {}.bak".format(conf_file, conf_file))
+                LOG.info("Restarting collectd service on node ({})"
+                         .format(node.fqdn))
+                node.run("systemctl restart collectd")
```

Comment:
Since this is within loop executing for each collectd conf file, it might be restarted repeatedly. Might be cleaner to set a boolean 'need_restart' and then restart outside of loop if set.